pkg/opentelemetry-mapping-go/otlp/logs/orchestrator.go

```diff
@@ -0,0 +1,534 @@
+func pullLogToManifestFromMap(k8sResource map[string]interface{}, logRecord plog.LogRecord, resource pcommon.Resource) (*agentmodel.Manifest, error) {
+	// Body is already parsed, just pass it to common logic
+	// Not a delete event in pull mode, not from watch
+	return BuildManifestFromK8sResource(k8sResource, resource, logRecord, false)
+}
+
+// BuildManifestFromK8sResource is the shared logic to convert a k8s resource map to a manifest.
+// This function is used by both watchLogToManifest and pullLogToManifest to ensure consistent
+// manifest creation regardless of the source mode.
+//
+// Parameters:
+//   - k8sResource: The Kubernetes resource as a map (already unmarshaled from JSON)
+//   - resource: OTLP resource containing additional metadata
+//   - logRecord: OTLP log record for extracting tags and attributes
+//   - isTerminated: true if this represents a deleted resource (watch mode only)
+func BuildManifestFromK8sResource(k8sResource map[string]interface{}, resource pcommon.Resource, logRecord plog.LogRecord, isTerminated bool) (*agentmodel.Manifest, error) {
+	// Check if the resource kind should be skipped for security reasons
+	kind, _ := k8sResource["kind"].(string)
+	group, _ := k8sResource["apiVersion"].(string)
+	if shouldSkipResourceKind(kind, group) {
+		return nil, fmt.Errorf("skipping unsupported resource kind: %s (contains sensitive data)", kind)
+	}
+
+	// Extract metadata
+	metadata, ok := k8sResource["metadata"].(map[string]interface{})
+	if !ok || metadata == nil {
+		return nil, fmt.Errorf("k8s resource missing metadata")
+	}
+
+	uid, _ := metadata["uid"].(string)
+	if uid == "" {
+		return nil, fmt.Errorf("k8s resource missing uid in metadata")
+	}
+
+	resourceVersion, _ := metadata["resourceVersion"].(string)
+	apiVersion, _ := k8sResource["apiVersion"].(string)
+
+	// Convert the Kubernetes resource to JSON bytes for the manifest content
+	content, err := json.Marshal(k8sResource)
+	if err != nil {
+		return nil, fmt.Errorf("failed to marshal k8s resource content: %w", err)
+	}
+
+	// Determine manifest type based on the Kubernetes resource kind
+	manifestType := getManifestType(kind)
+
+	// Build tags from resource and log record attributes
+	tags := buildTags(resource, logRecord)
+
+	// Create the manifest
+	manifest := &agentmodel.Manifest{
+		Type:            int32(manifestType),
+		ResourceVersion: resourceVersion,
+		Uid:             uid,
+		Content:         content,
+		ContentType:     "application/json",
+		Version:         "v1",
+		Tags:            tags,
+		IsTerminated:    isTerminated,
+		ApiVersion:      apiVersion,
+		Kind:            kind,
+	}
+	return manifest, nil
+}
+
+// ToManifestPayload creates a CollectorManifest payload from a list of manifests.
+func ToManifestPayload(manifests []*agentmodel.Manifest, hostName, clusterName, clusterID string, logger *zap.Logger) *agentmodel.CollectorManifest {
+	version, err := version.Agent()
```

Comment:
Were you able to test this yet in the OSS Collector? I'm not sure if this would work as expected but not too familiar with this package